hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/validators/ConsensusCustomFeesValidator.java

```diff
@@ -0,0 +1,141 @@
+     * @param tokenStore         The token store.
+     * @param customFees         The custom fees to validate.
+     * @param expiryValidator    The expiry validator to use (for fee collector accounts)
+     */
+    public void validate(
+            @NonNull final ReadableAccountStore accountStore,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final ReadableTokenStore tokenStore,
+            @NonNull final List<ConsensusCustomFee> customFees,
+            @NonNull final ExpiryValidator expiryValidator) {
+        requireNonNull(accountStore);
+        requireNonNull(tokenRelationStore);
+        requireNonNull(tokenStore);
+        requireNonNull(customFees);
+        requireNonNull(expiryValidator);
+
+        for (final var fee : customFees) {
+            // Validate the fee collector account is in a usable state
+            getIfUsableForAliasedId(
+                    fee.feeCollectorAccountIdOrElse(AccountID.DEFAULT),
+                    accountStore,
+                    expiryValidator,
+                    INVALID_CUSTOM_FEE_COLLECTOR);
+
+            final var isSpecified = fee.hasFixedFee();
+            validateTrue(isSpecified, CUSTOM_FEE_NOT_FULLY_SPECIFIED);
+            validateFixedFee(fee, tokenRelationStore, tokenStore);
+        }
+    }
+
+    private void validateFixedFee(
+            @NonNull final ConsensusCustomFee fee,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var fixedFee = fee.fixedFeeOrThrow();
+        validateTrue(fixedFee.amount() > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        if (fixedFee.hasDenominatingTokenId()) {
+            validateExplicitTokenDenomination(
+                    fee.feeCollectorAccountId(),
+                    fixedFee.denominatingTokenId(),
+                    fixedFee.amount(),
+                    tokenRelationStore,
+                    tokenStore);
+        }
+    }
+
+    /**
+     * Validate explicitly set token denomination for custom fees.
+     *
+     * @param feeCollectorNum The fee collector account number.
+     * @param tokenNum The token number used for token denomination.
+     * @param feeAmount The fee amount.
+     * @param tokenRelationStore The token relation store.
+     * @param tokenStore The token store.
+     */
+    private void validateExplicitTokenDenomination(
+            @NonNull final AccountID feeCollectorNum,
+            @NonNull final TokenID tokenNum,
+            final long feeAmount,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var denomToken = getIfUsable(tokenNum, tokenStore, REQUIRE_NOT_PAUSED, INVALID_TOKEN_ID_IN_CUSTOM_FEES);
+        validateFalse(
+                TokenSupplyType.FINITE.equals(denomToken.supplyType()) && feeAmount > denomToken.maxSupply(),
```

Comment:
It will easier to read 
```
if(denomToken.supplyType().equals(TokenSupplyType.FINITE)){
            validateTrue(feeAmount > denomToken.maxSupply(), AMOUNT_EXCEEDS_TOKEN_MAX_SUPPLY);
        }
```